### PR TITLE
Use `thiserror` rather than `eyre` in `mijia` crate

### PR DESCRIPTION
**Pull request by qwandor**
It seems more appropriate for a library.

### CODE REVIEW

**Review by alsuren — APPROVED** [review]
See what this does to the backtraces and then ship it? Would be nice to know whether converting from box dyn Error to thiserror to eyre throws stack frames away. It may be that the `backtrace` crate registers itself in a magical way that makes everything get preserved, but I'm also okay with losing a few frames of dbus-rs as long as we can still report the place where our thiserror is being constructed.

File: mijia/src/bluetooth.rs [hunk 1]
```diff
@@ -5,14 +5,35 @@ use core::future::Future;
...
+    /// There was an error talking to the BlueZ daemon over D-Bus.
+    #[error(transparent)]
+    DbusError(#[from] dbus::Error),
+}
+
+/// Error type for futures representing tasks spawned by this crate.
+#[derive(Error, Debug)]
+pub enum SpawnError {
+    #[error("D-Bus connection lost: {0}")]
+    DbusConnectionLost(String),
```
[the remark this answers is not in the document]
Comment (qwandor) [reply mijia/src/bluetooth.rs]: Done.

File: mijia/src/decode/comfort_level.rs [hunk 2]
```diff
@@ -17,10 +18,8 @@ pub struct ComfortLevel {
 }
 
 impl ComfortLevel {
-    pub(crate) fn decode(value: &[u8]) -> Result<ComfortLevel, eyre::Report> {
-        if value.len() != 6 {
-            bail!("Wrong length {} for comfort level", value.len());
-        }
+    pub(crate) fn decode(value: &[u8]) -> Result<ComfortLevel, DecodeError> {
```
[the remark this answers is not in the document]
Comment (qwandor) [reply mijia/src/decode/comfort_level.rs:21]: That would mean that the decoding logic is part of the public API of the crate, which I'm not sure we want. I was considering adding some private traits, but it doesn't really give any advantage, and prevents some of the slight differences in signatures we currently have (e.g. encoding `TemperatureUnit` can't fail, so doesn't return a `Result`).

**Comment by qwandor** [comment]
Backtraces go up to the point that `mijia-homie` calls into `mijia` library methods and have the error which they return as cause, which I think is enough. Unfortunately `thiserror` only supports backtraces with nightly Rust, not with the `backtrace` crate. Added #85 to track this.

**Review by alsuren — APPROVED** [review]

**Pull request merged**